src/reboost/hpge/psd.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import logging
+
+import awkward as ak
+import numpy as np
+from lgdo import Array
+
+log = logging.getLogger(__name__)
+
+
+def r90(edep: np.ndarray, xloc: np.ndarray, yloc: np.ndarray, zloc: np.ndarray):
+    """Computes R90 for each hit in a ged.
+
+    Parameters
+    ----------
+    edep
+        awkward array of energy
+    xloc
+        awkward array of x coordinate position
+    yloc
+        awkward array of y coordinate position
+    zloc
+        awkward array of z coordinate position
+
+    Returns
+    -------
+    r90
+    """
+    tot_energy = ak.sum(edep, axis=-1, keepdims=True)
+
+    # Compute energy-weighted mean positions
+    mean_x = ak.sum(edep * xloc, axis=-1, keepdims=True) / tot_energy
+    mean_y = ak.sum(edep * yloc, axis=-1, keepdims=True) / tot_energy
+    mean_z = ak.sum(edep * zloc, axis=-1, keepdims=True) / tot_energy
+
+    xdiff = xloc - mean_x
+    ydiff = yloc - mean_y
+    zdiff = zloc - mean_z
+
+    xdiff = xdiff * xdiff
+    ydiff = ydiff * ydiff
+    zdiff = zdiff * zdiff
+
+    # Compute distance of each edep to the weighted mean
+    dist = np.sqrt(xdiff + ydiff + zdiff)
+
+    # Sort distances and corresponding edep within each event
+    sorted_indices = ak.argsort(dist, axis=-1)
+    sorted_dist = dist[sorted_indices]
+    sorted_edep = edep[sorted_indices]
+
+    def cumsum(layout, **kwargs):  # noqa: ARG001
+        if layout.is_numpy:
+            return ak.contents.NumpyArray(np.cumsum(layout.data))  # noqa: disallow-caps
```

Comment:
```suggestion
    def cumsum(layout, **_kwargs):
        if layout.is_numpy:
            return ak.contents.NumpyArray(np.cumsum(layout.data))
```

the second one should have been resolved by merging `main`